packages/jest-cli/src/TestRunner.js

```diff
@@ -280,30 +278,91 @@ class TestRunner {
     return Promise.race([runAllTests, onInterrupt]).then(cleanup, cleanup);
   }
 
+  _shouldAddDefaultReporters(reporters?: Array<ReporterConfig>): boolean {
+    return !reporters || reporters.indexOf(DEFAULT_REPORTER_LABEL) !== -1;
+  }
+
   _setupReporters() {
-    const {collectCoverage, expand, notify, verbose} = this._globalConfig;
+    const {collectCoverage, notify, reporters} = this._globalConfig;
 
-    this.addReporter(
-      verbose
-        ? new VerboseReporter({expand})
-        : new DefaultReporter({verbose: !!verbose}),
-    );
+    const isDefault = this._shouldAddDefaultReporters(reporters);
+
+    if (isDefault) {
+      this._setupDefaultReporters(this._globalConfig);
+    }
+
+    if (reporters && Array.isArray(reporters)) {
+      this._addCustomReporters(reporters);
+    }
 
     if (collectCoverage) {
       // coverage reporter dependency graph is pretty big and we don't
       // want to require it if we're not in the `--coverage` mode
       const CoverageReporter = require('./reporters/CoverageReporter');
       this.addReporter(
-        new CoverageReporter({maxWorkers: this._options.maxWorkers}),
+        new CoverageReporter(this._globalConfig, {
+          maxWorkers: this._options.maxWorkers,
+        }),
       );
     }
 
-    this.addReporter(new SummaryReporter(this._options));
     if (notify) {
       this.addReporter(new NotifyReporter(this._options.startRun));
     }
   }
 
+  _setupDefaultReporters(config: GlobalConfig) {
+    this.addReporter(
+      config.verbose
+        ? new VerboseReporter(config)
+        : new DefaultReporter(this._globalConfig),
+    );
+
+    this.addReporter(
+      new SummaryReporter(this._globalConfig, {
+        pattern: this._options.pattern,
+        testNamePattern: this._options.testNamePattern,
+        testPathPattern: this._options.testPathPattern,
+      }),
+    );
+  }
+
+  _addCustomReporters(reporters: Array<ReporterConfig>) {
+    const customReporter = reporters.filter(reporter => reporter !== 'default');
+
+    customReporter.forEach((reporter, index) => {
+      const {options, path} = this._getReporterProps(reporter);
+
+      try {
+        const Reporter = require(path);
+        this.addReporter(new Reporter(this._globalConfig, options));
+      } catch (error) {
+        console.error(
+          chalk.red(
+            'An error occured while adding the reporter at path ' + path,
+          ),
+        );
+        throw error;
+      }
+    });
+  }
+
+  /**
+   * Get properties of a reporter in an object
+   * to make dealing with them less painful.
+   */
+  _getReporterProps(reporter: ReporterConfig): Object {
```

Comment:
I think you can remote `Object` here because it is a private field. If not, then please type it properly :)